Builds break because System.Net.Http NuGet and assembly reference do not agree

### Description

The net72 TargetFramework build of FluentAssertions in your NuGet package distribution is broken. The nuspec states this dependency:

```xml
        <dependency id="System.Net.Http" version="4.3.4" exclude="Build,Analyzers" />
```

This dependency produces a System.Net.Http reference with assembly version 4.1.1.3:
![image](https://user-images.githubusercontent.com/3548/216731915-f75fc360-f9e0-4c9e-ad8a-a6e24ab6b710.png)

Yet your FluentAssertions assembly itself references a _higher_ assembly version of 4.2.0.0 for System.Net.Http:
![image](https://user-images.githubusercontent.com/3548/216731955-74d1fd48-afaa-48d9-affa-bef8d07762bd.png)

This causes some consuming MSBuild projects to fail at the compiler with this error:

> CSC : error CS1705: Assembly 'FluentAssertions' with identity 'FluentAssertions, Version=6.8.0.0, Culture=neutral, PublicKeyToken=33f2691a05b67b6a' uses 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Net.Http' with identity 'System.Net.Http, Version=4.1.1.3, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

In fact no System.Net.Http package version offers an assembly version of 4.2.0.0. This leaves two options for fixing:

## Option 1

Actually compile against the System.Net.Http that your nuspec documents, giving your assembly reference a 4.1.1.3 assembly version instead of 4.2.0.0. This would remove the conflict in the compiler and resolve the error for your users.

## Option 2

Because the 4.2.0.0 assembly version comes from the .NET Framework targeting pack rather than nuget, remove the System.Net.Http nuget package reference from your net47 build and instead add a `frameworkAssemblies` reference to your nuspec:

```diff
    <dependencies>
      <group targetFramework=".NETFramework4.7">
-       <dependency id="System.Net.Http" version="4.3.4" exclude="Build,Analyzers" />
        <dependency id="System.Threading.Tasks.Extensions" version="4.5.0" exclude="Build,Analyzers" />
      </group>
 ...
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="System.Configuration" targetFramework=".NETFramework4.7" />
      <frameworkAssembly assemblyName="System.Data" targetFramework=".NETFramework4.7" />
      <frameworkAssembly assemblyName="System.Xml" targetFramework=".NETFramework4.7" />
      <frameworkAssembly assemblyName="System.Xml.Linq" targetFramework=".NETFramework4.7" />
+     <frameworkAssembly assemblyName="System.Net.Http" targetFramework=".NETFramework4.7" />
    </frameworkAssemblies>
```



### Reproduction Steps

Unfortunately all the repro cases I have are in internal projects, and I haven't been able to repro on a File-> New Project to share with you.
I'm hoping that you can see from the Description how this is (at least arguably) faulty though and how the fix should hopefully be straightforward. We have a lot of projects that use FluentAssertions and working around this bug is quite tedious. 

### Expected behavior

The references that are produced by the nuget package in our consuming projects should match or exceed the assembly references in FluentAssertions.dll.

### Actual behavior

 some consuming MSBuild projects to fail at the compiler with this error:

> CSC : error CS1705: Assembly 'FluentAssertions' with identity 'FluentAssertions, Version=6.8.0.0, Culture=neutral, PublicKeyToken=33f2691a05b67b6a' uses 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Net.Http' with identity 'System.Net.Http, Version=4.1.1.3, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

--- a/Src/FluentAssertions/FluentAssertions.csproj
+++ b/Src/FluentAssertions/FluentAssertions.csproj
@@ -91,7 +91,10 @@
   </ItemGroup>
   <ItemGroup Condition="'$(TargetFramework)' == 'net47'">
     <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.0" />
-    <PackageReference Include="System.Net.Http" Version="4.3.4" />
+    <TfmSpecificFrameworkAssemblyReferences Include="System.Net.Http">
+      <TargetFramework>$(TargetFramework)</TargetFramework>
+    </TfmSpecificFrameworkAssemblyReferences>
+    <Reference Include="System.Net.Http" />
     <Reference Include="System.Configuration" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />

--- a/Tests/Benchmarks/Benchmarks.csproj
+++ b/Tests/Benchmarks/Benchmarks.csproj
@@ -15,5 +15,6 @@
   </ItemGroup>
   <ItemGroup>
     <Reference Include="System.Data.DataSetExtensions" Condition="'$(TargetFramework)' == 'net472'" />
+    <Reference Include="System.Net.Http" Condition="'$(TargetFramework)' == 'net472'" />
   </ItemGroup>
 </Project>

--- a/docs/_pages/releases.md
+++ b/docs/_pages/releases.md
@@ -16,6 +16,8 @@ sidebar:
 
 * `BeEquivalentTo` no longer crashes on fields hiding base-class fields - [#1990](https://github.com/fluentassertions/fluentassertions/pull/1990)
 
+* Fixed System.Net.Http dependency declaration for net47 target framework to be a framework dependency instead of a nuget dependency - [#2122](https://github.com/fluentassertions/fluentassertions/pull/2122)
+
 ## 6.9.0
 
 ### What's new

